backend/entities/office_hours/oh_event_entity.py

```diff
@@ -50,10 +52,72 @@ class OfficeHoursEventEntity(EntityBase):
     )
 
     # NOTE: Unidirectional relationship to Room
-    location_id: Mapped[int] = mapped_column(ForeignKey("room.id"), nullable=False)
-    location: Mapped["RoomEntity"] = relationship("RoomEntity")
+    room_id: Mapped[int] = mapped_column(ForeignKey("room.id"), nullable=False)
+    room: Mapped["RoomEntity"] = relationship("RoomEntity")
 
     # NOTE: One-to-many relationship of OfficeHoursEvent to tickets
     tickets: Mapped[list["OfficeHoursTicketEntity"]] = relationship(
-        back_populates="event", cascade="all, delete"
-    )
+        back_populates="event", cascade="all, delete")
+
+    @classmethod
+    def from_model(cls, model: OfficeHoursEvent) -> Self:
+        """
+        Class method that converts an `OfficeHoursEvent` model into a `OfficeHoursEventEntity`
+
+        Parameters:
+            - model (OfficeHoursEvent): Model to convert into an entity
+        Returns:
+            OfficeHoursEventEntity: Entity created from model
+        """
+        return cls(
+            id=model.id,
+            office_hours_section_id=model.office_hours_section_id,
+            room_id=model.room_id,
+            type=model.type,
+            description=model.description,
+            location_description=model.location_description,
+            date=model.event_date,
```

Comment:
I believe this should be `date=model.date` since the field that represents `date` in the `OfficeHoursEvent` class is named that! I only discovered this when trying to add test data for this table!

Reply:
Maybe we could just go back to naming it from `event_date` to `date`. I think we wanted to name it `event_date` at first since we wanted to avoid confusion with the `date` type

Reply:
Or we can change all `date` fields in our models to `event_date`

Reply:
These were all great catches, thank you!! This has been updated now from event_date to date.